lib/chef/resource/apt_repository.rb

```diff
@@ -378,7 +379,7 @@ def make_ppa_url(ppa)
         def build_repo(uri, distribution, components, trusted, arch, add_src = false)
           uri = make_ppa_url(uri) if is_ppa_url?(uri)
 
-          uri = URI.escape(uri)
+          uri = CGI.escape(uri)
```

Comment:
this one is problematic because its ripping the escaping across the entire uri, the other ones are specifically targeting the user/password/path components.

Reply:
fixed with Addressable